tensorflow/contrib/tensorrt/resources/trt_allocator.cc

```diff
@@ -37,8 +37,22 @@ void TRTCudaAllocator::free(void* memory) { cudaFree(memory); }
 
 void* TRTDeviceAllocator::allocate(uint64_t size, uint64_t alignment,
                                    uint32_t flags) {
+  // WAR for allocator alignment requirement
+  alignment = 512;
```

Comment:
Why do we need to set it to 512?

Reply:
Temporary WAR for TRT allocator. Some cuda API calls require allocated gpu memory to abide to cudaDeviceProp::textureAlignment. The specific call that's causing the error: https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TEXTURE__OBJECT.html

Advised by the TRT team to just override it to 512.

Reply:
Thanks, would you help to add this as comments and reference #20856?